### DISCUSSION TITLE
Feature: RTSP Stream Auto Recovery

### DISCUSSION THREAD

**Discussion by jsammarco** [ultralytics/ultralytics#3929] [Ideas]
Hi All,
I really needed this feature so I modified the update function in the stream_loaders.py to handle a crashed stream and attempt to auto reconnect. I tested this with an IP camera and unplugged, plugged it back in and it fully recovered.
I hope this helps and it gets added to MAIN branch.
Thanks
Joe

```
    def update(self, i, cap, stream):
        """Read stream `i` frames in daemon thread."""
        n, f = 0, self.frames[i]  # frame number, frame array
        while cap.isOpened() and n < f:
            n += 1
            try:
                cap.grab()  # .read() = .grab() followed by .retrieve()
                if n % self.vid_stride == 0:
                    success, im = cap.retrieve()
                    if success:
                        self.imgs[i] = im
                    else:
                        LOGGER.warning('WARNING ⚠️ Video stream unresponsive, please check your IP camera connection.')
                        self.imgs[i] = np.zeros_like(self.imgs[i])
                        LOGGER.warning('WARNING ⚠️ Closing IP camera connection.')
                        cap.release()  # close stream if signal was lost
                        time.sleep(3.0)
                        LOGGER.warning('WARNING ⚠️ Re-Opening IP camera connection. '+stream)
                        cap.open(stream)  # re-open stream if signal was lost
            except cv2.error as e:
                LOGGER.warning('WARNING ⚠️ Video stream unresponsive, please check your IP camera connection.')
                self.imgs[i] = np.zeros_like(self.imgs[i])
                LOGGER.warning('WARNING ⚠️ Closing IP camera connection.')
                cap.release()  # close stream if signal was lost
                time.sleep(3.0)
                LOGGER.warning('WARNING ⚠️ Re-Opening IP camera connection. '+stream)
                cap.open(stream)  # re-open stream if signal was lost
            time.sleep(0.0)  # wait time
```

**Comment by glenn-jocher** [comment]
@jsammarco hi Joe,

Thank you for sharing your enhancement for RTSP stream auto-recovery with the community! It's great to see active contributions that can help improve the robustness of YOLOv8's streaming capabilities.

We appreciate the time and effort you've put into developing this feature. For potential inclusion in the main branch, we typically review the code's functionality, compatibility, and impact on the existing codebase. I encourage you to submit a pull request with your proposed changes. This way, our team can properly evaluate and test your code in a variety of scenarios to ensure it meets the project's standards.

Once again, thanks for your initiative, and we look forward to seeing your pull request! If you have any further questions or need assistance, feel free to reach out.
